extension/observer/cfgardenobserver/config_test.go

```diff
@@ -26,30 +27,181 @@ func TestLoadConfig(t *testing.T) {
-				Endpoint:        "unix:///var/vcap/data/garden/garden.sock",
-				RefreshInterval: 1 * time.Minute,
+				RefreshInterval:   1 * time.Minute,
+				CacheSyncInterval: 5 * time.Minute,
+				IncludeAppLabels:  false,
+				Garden: GardenConfig{
+					Endpoint: "/var/vcap/data/garden/garden.sock",
+				},
 			},
 		},
 		{
 			id: component.NewIDWithName(metadata.Type, "all_settings"),
 			expected: &Config{
-				Endpoint:        "unix:///var/vcap/data/garden/custom.sock",
-				RefreshInterval: 20 * time.Second,
+				RefreshInterval:   20 * time.Second,
+				CacheSyncInterval: 5 * time.Second,
+				IncludeAppLabels:  true,
+				Garden: GardenConfig{
+					Endpoint: "/var/vcap/data/garden/custom.sock",
+				},
+				CloudFoundry: CfConfig{
+					Endpoint: "https://api.cf.mydomain.com",
+					AuthType: "user_pass",
+					Username: "myuser",
+					Password: "mypass",
+				},
+			},
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "user_pass"),
+			expected: &Config{
+				Garden: GardenConfig{
+					Endpoint: "/var/vcap/data/garden/garden.sock",
+				},
+				RefreshInterval:   1 * time.Minute,
+				CacheSyncInterval: 5 * time.Minute,
+				IncludeAppLabels:  true,
+				CloudFoundry: CfConfig{
+					Endpoint: "https://api.cf.mydomain.com",
+					AuthType: "user_pass",
+					Username: "myuser",
+					Password: "mypass",
+				},
+			},
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "client_credentials"),
+			expected: &Config{
+				Garden: GardenConfig{
+					Endpoint: "/var/vcap/data/garden/garden.sock",
+				},
+				RefreshInterval:   1 * time.Minute,
+				CacheSyncInterval: 5 * time.Minute,
+				IncludeAppLabels:  true,
+				CloudFoundry: CfConfig{
+					Endpoint:     "https://api.cf.mydomain.com",
+					AuthType:     "client_credentials",
+					ClientID:     "myclientid",
+					ClientSecret: "myclientsecret",
+				},
+			},
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "token"),
+			expected: &Config{
+				Garden: GardenConfig{
+					Endpoint: "/var/vcap/data/garden/garden.sock",
+				},
+				RefreshInterval:   1 * time.Minute,
+				CacheSyncInterval: 5 * time.Minute,
+				IncludeAppLabels:  true,
+				CloudFoundry: CfConfig{
+					Endpoint:     "https://api.cf.mydomain.com",
+					AuthType:     "token",
+					AccessToken:  "myaccesstoken",
+					RefreshToken: "myrefreshtoken",
+				},
 			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.id.String(), func(t *testing.T) {
-			cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
-			require.NoError(t, err)
-			factory := NewFactory()
-			cfg := factory.CreateDefaultConfig()
-			sub, err := cm.Sub(tt.id.String())
-			require.NoError(t, err)
-			require.NoError(t, sub.Unmarshal(cfg))
-
+			cfg := loadConfig(t, tt.id)
 			assert.NoError(t, component.ValidateConfig(cfg))
 			assert.Equal(t, tt.expected, cfg)
 		})
 	}
 }
+
+func TestConfigValidate(t *testing.T) {
+	cases := []struct {
+		reason string
+		cfg    Config
+		msg    string
```

Comment:
Just a thought: I see some value in checking specific error messages here, but I'm also concerned that this will cause a lot of noise when needing to update error messages for clarity etc. We could simply switch to a boolean that shows whether we expect and error or not. Your call, whatever you think makes more sense 👍

Reply:
Had the same thought, but with the different conditionals it's pretty easy to get an error by getting into the wrong branch, so I think we can keep the messages for now and remove later if it turns out to be noisy.